drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -121,6 +186,9 @@ local function assign_child_profile(device, child_ep)
         id = math.max(id, dt.device_type_id)
       end
       profile = device_type_profile_map[id]
+      for _, attr in pairs(device_type_attribute_map[id]) do
```

Comment:
Is this the best place to add the subscriptions? It looks like we're going to be doing that before the child devices are created since this is the function that determines the profile to use when creating the child device. Why are we not waiting until the child device is created to add the subscription so that it's ready when the device starts sending subscription reports?

Reply:
I thought this was a logical place for this since we are already determining the device type here. If we add the subscriptions here before the child devices are created, then wouldn't they be ready once the device starts sending the subscription reports?

Reply:
I could be mistaken here, but I think that if the subscriptions are added before the devices are created then the initial priming report will not result in any capability events being generated since the devices won't exist yet.

Reply:
Hi Tom, I missed this message. I was under the impression that the subscriptions are all made through the parent device, so would this still be a concern if that is indeed the case?

Reply:
After discussing, the subscriptions were moved to `device_init` so that they will be subscribed to each time the device is reinitialized.